liquibase-core/src/main/java/liquibase/sqlgenerator/core/UpdateChangeSetChecksumGenerator.java

```diff
@@ -34,11 +36,20 @@ public Sql[] generateSql(UpdateChangeSetChecksumStatement statement, Database da
                     .setWhereClause(database.escapeObjectName("ID", LiquibaseColumn.class) + " = ? " +
                             "AND " + database.escapeObjectName("AUTHOR", LiquibaseColumn.class) + " = ? " +
                             "AND " + database.escapeObjectName("FILENAME", LiquibaseColumn.class) + " = ?")
-                    .addWhereParameters(changeSet.getId(), changeSet.getAuthor(), changeSet.getFilePath());
+                    .addWhereParameters(changeSet.getId(), changeSet.getAuthor(), getFilePathToUseInWhereClause(changeSet));
```

Comment:
If we always use the now filled changeSet.getStoredFilePath() instead of changeSet.getFilePath() would this problem be solved?